addons/mail/static/src/discuss/core/web/channel_member_list_patch.js

```diff
@@ -0,0 +1,24 @@
+import { ChannelMemberList } from "@mail/discuss/core/common/channel_member_list";
+import { AvatarCardPopover } from "@mail/discuss/web/avatar_card/avatar_card_popover";
+
+import { usePopover } from "@web/core/popover/popover_hook";
+import { patch } from "@web/core/utils/patch";
+
+patch(ChannelMemberList.prototype, {
+    setup() {
+        super.setup(...arguments);
+        this.avatarCard = usePopover(AvatarCardPopover);
+    },
+    onClickAvatar(ev, member) {
+        if (!this.canOpenChatWith(member)) {
```

Comment:
Why would not being able to open chat prevent from opening avatar card?

Reply:
the `onClickAvatar` is a little bit confusing indeed. but the idea is for a public page and for guests, it is not able to open chat, which is the same requirement as opening the avatar card. As talked to the po, we don't want to show an empty card with no info inside

Reply:
Not the best limitation but ok for now then.